src/region/RegionCreator.tsx

```diff
@@ -0,0 +1,178 @@
+    };
+
+    handleMouseDown = ({ buttons, clientX, clientY }: React.MouseEvent): void => {
+        if (buttons !== MOUSE_PRIMARY) {
+            return;
+        }
+
+        this.startDraw({
+            x: clientX,
+            y: clientY,
+        });
+    };
+
+    handleMouseMove = ({ buttons, clientX, clientY }: MouseEvent): void => {
+        const { isDrawing } = this.state;
+
+        if (buttons !== MOUSE_PRIMARY || !isDrawing) {
+            return;
+        }
+
+        this.updateDraw({
+            x: clientX,
+            y: clientY,
+        });
+    };
+
+    handleMouseUp = (): void => {
+        this.stopDraw();
+    };
+
+    handleTouchCancel = (): void => {
+        this.stopDraw();
+    };
+
+    handleTouchEnd = (): void => {
+        this.stopDraw();
+    };
+
+    handleTouchMove = ({ targetTouches }: React.TouchEvent): void => {
+        this.updateDraw({
+            x: targetTouches[0].clientX,
+            y: targetTouches[0].clientY,
+        });
+    };
+
+    handleTouchStart = ({ targetTouches }: React.TouchEvent): void => {
+        this.startDraw({
+            x: targetTouches[0].clientX,
+            y: targetTouches[0].clientY,
+        });
+    };
+
+    startDraw(position: Position): void {
+        this.setState({
+            isDrawing: true,
+            position: this.getPosition(position),
+        });
+    }
+
+    stopDraw(): void {
+        const { onDone } = this.props;
+
+        this.setState({
+            isDrawing: false,
+        });
+
+        onDone();
+    }
+
+    updateDraw({ x, y }: Position): void {
+        const { onDraw } = this.props;
+        const { position: position1 } = this.state;
+        const { current: creatorRef } = this.creatorRef;
+
+        if (!position1 || !creatorRef) {
+            return;
+        }
+
+        // Get the maximum x and y coordinates for any given rectangle based on the size of the stage
+        const { height, width } = creatorRef.getBoundingClientRect();
+        const MAX_X = Math.max(0, width - MIN_X);
+        const MAX_Y = Math.max(0, height - MIN_Y);
+
+        // Get the first position from mousedown and the current position
+        const { x: x1, y: y1 } = position1;
+        const { x: x2, y: y2 } = this.getPosition({ x, y });
+
+        // Set the origin x/y to the lowest value and the target x/y to the highest to avoid negative height/width
+        const originX = Math.min(Math.max(MIN_X, x1 < x2 ? x1 : x2), MAX_X);
+        const originY = Math.min(Math.max(MIN_Y, y1 < y2 ? y1 : y2), MAX_Y);
+        const targetX = Math.min(Math.max(MIN_X, x2 > x1 ? x2 : x1), MAX_X);
+        const targetY = Math.min(Math.max(MIN_Y, y2 > y1 ? y2 : y1), MAX_Y);
+
+        onDraw({
+            height: Math.max(MIN_SIZE, targetY - originY),
+            width: Math.max(MIN_SIZE, targetX - originX),
+            x: originX,
+            y: originY,
+        });
+    }
+
+    render(): JSX.Element | null {
```

Comment:
Is the null needed?

Reply:
Nope, good catch. I think I was returning null in a guard clause at some point.

Reply:
I'll address this in the next PR, if that's cool.